pilot/pkg/networking/core/v1alpha3/listener.go

```diff
@@ -269,6 +291,26 @@ func (configgen *ConfigGeneratorImpl) buildSidecarListeners(env *model.Environme
 					},
 				},
 			}
+			mutable := &plugin.MutableObjects{
+				Listener:     ipTablesListener,
+				FilterChains: make([]plugin.FilterChain, len(ipTablesListener.FilterChains)),
+			}
+
+			for _, p := range configgen.Plugins {
+				if err := p.OnOutboundListener(pluginParams, mutable); err != nil {
+					log.Warn(err.Error())
+				}
+			}
+			if len(mutable.FilterChains) > 0 && len(mutable.FilterChains[0].TCP) > 0 {
+				filters := append([]listener.Filter{}, mutable.FilterChains[0].TCP...)
+				filters = append(filters, filter)
+				ipTablesListener.FilterChains = []listener.FilterChain{
+					{
+						Filters: filters,
+					},
+				}
+			}
+			builder.virtualListener = ipTablesListener
```

Comment:
this is insufficient as the listener builder does not use this code.

Reply:
yeah, I will add a commit to accommodate for the new listener builder mechanism.

Reply:
Fixed in the code.